futures/src/tests.rs

```diff
@@ -0,0 +1,37 @@
+use futures::future::Future;
+use futures::Stream;
+use tokio_core::reactor::Core;
+use tokio_core::net::TcpStream;
+use super::client::{self, ConnectionOptions};
+use super::types::FieldTable;
+
+#[test]
+fn qos_and_queue() {
```

Comment:
I don't think adding a whole new test is relevant, just hook up the new method in the existing test